draco/synthesis/noise.py

```diff
@@ -60,6 +61,35 @@ def process(self, data):
 
         return data
 
+class GaussianInPlace(task.SingleTask):
+    '''Generates a Gaussian distributed noise dataset using the
+    the noise estimates of an existing dataset.
+
+    Attributes
+    ----------
+    Related to distinguishing different container types
+    We need to be able to support both draco and CHIME timestream types
+    '''
+
+    def process(self, data):
+        '''Generates a noise dataset, given the provided dataset's visibility weights'''
+        data.redistribute("freq")
+
+        vis = data.vis[:]
+        vis_weight = np.copy(data.weight)
+
+        # vis_weight elements are inverse variances
+        # we want the standard deviation
+        vis_weight_std = np.sqrt(1. / vis_weight)
```

Comment:
It seems like `invert_no_zero` returns a "zero" in the cases where it receives a "zero". So that would mean a variance of 0, and the random number generator will return a 0. 

I think that is reasonable behaviour, though if we had some concept of "`None`" that would probably be better.